docs/framework/chat-assistant.mdx

```diff
@@ -2,153 +2,325 @@
+  </Step>
+</Steps>
+
+The project editor should now look like this:
+
+![Project editor with components laid out](/framework/images/tutorial/chat/chat_assistant_2c.png)
+
+
+## Add the back-end code
+
+With the UI laid out, it's time to work on the logic behind it. 
+
+Switch to your code editor and open the `main.py` file in your project's directory. This file was automatically generated; you'll update the code in it to define the behavior of your app.
+
+<Steps>
+  <Step title="Import libraries and load the Writer Framework API key">
+  You should see the following at the start of the file:
+
+  ```python
+  import writer as wf
+  import writer.ai
+  ```
+
+  Replace that code with the following:
+
+  ```python
+  import os
+  from dotenv import load_dotenv
+  import writer as wf
+  import writer.ai
+
+  # Load environment variables from .env file
+  load_dotenv()
+
+  # Set the API key
+  wf.api_key = os.getenv("WRITER_API_KEY")
+  ```
+  
+  This code imports the libraries that the application will need and then reads your Writer Framework API key from the `.env` file in your project's directory.
+  </Step>
+
+  <Step title="Initialize the application">
+  The next step is to set the application's initial state. Find this code...
+
+  ```python
+  # Welcome to Writer Framework! 
+  # This template is a starting point for your AI apps.
+  # More documentation is available at https://dev.writer.com/framework
+
+
+  # Initialise the state
+  wf.init_state({
+      "my_app": {
+          "title": "AI STARTER"
+      },
+  })
+  ```
+
+  ...and replace it with the following:
+
+  ```python
+  # Initialize the state
+  wf.init_state({
+      "conversation": writer.ai.Conversation(),
+      "my_app": {
+          "title": "CHAT ASSISTANT"
+      },
+  })
+  ```
+
+  The Writer Framework's `init_state()` method sets the initial value of `state`, a dictionary containing values that define the state of the application. The key-value pairs in `state` are how you store values used by your app and how you pass data between the back-end code and the UI.
+
+  The code above sets the initial value of `state` so that it has two key-value pairs:
+
+  - `conversation`: An object that keeps a record of the conversation that the user is having with the LLM. You'll bind its value to the **Chatbot** component soon.
+  - `my_app`: A dictionary containing values that define the application's appearance. This dictionary has a single key-value pair, `title`, which defines the text that appears as the application's title in the **Header**.
+  </Step>
+
+  <Step title="Create a handler to respond to the user's input">
+  The application needs a function to handle incoming chat messages. Add the following to the end of `main.py`:
+
+  ```python
+  def generate_completion(state, payload):
+    print(f"Here's what the user entered: {payload['content']}")
```

Comment:
I love all of the print statements here for debugging, but you can just call it out in the para below? Something about how there are some optional print statements in there to help with debugging, feel free to remove them.

Reply:
Updated.